linode_api4/linode_client.py

```diff
@@ -733,6 +733,115 @@ def users(self, *filters):
         """
         return self.client._get_and_filter(User, *filters)
 
+    def logins(self):
+        """
+        Returns a collection of successful logins for all users on the account during the last 90 days.
+        """
+
+        return self.client._get_and_filter(Login)
+
+    def maintenance(self):
+        """
+        Returns a collection of Maintenance objects for any entity a user has permissions to view. Cancelled Maintenance objects are not returned.
+        """
+
+        result = self.client.get(
+            "{}/maintenance".format(Account.api_endpoint), model=self
+        )
+
+        return [MappedObject(**r) for r in result["data"]]
+
+    def payment_methods(self):
+        """
+        Returns a  list of Payment Methods for this Account.
+        """
+
+        return self.client._get_and_filter(PaymentMethod)
+
+    def add_payment_method(self, data, is_default, type):
+        """
+        Adds a Payment Method to your Account with the option to set it as the default method.
+        """
+
+        if type != "credit_card":
+            raise ValueError("Unknown Payment Method type: {}".format(type))
+
+        if (
+            "card_number" not in data["data"]
+            or "expiry_month" not in data["data"]
+            or "expiry_year" not in data["data"]
+            or "cvv" not in data["data"]
+            or not data["data"]
+        ):
+            raise ValueError("Invalid credit card info provided")
+
+        params = {"data": data, "type": type, "is_default": is_default}
+
+        self.client.post(
+            "{}/payment-methods".format(Account.api_endpoint),
+            model=self,
+            data=params,
+        )
```

Comment:
I just checked this [recent change](https://github.com/linode/linode_api4-python/pull/211/files) and realized that we probably need to handle the error resp in another way. Looking at that PR, when the `errors` returns from raw json object, in the `_api_call` it will be parsed and returned as an error message instead of the raw json having the `errors` field. Sorry for the confusion!

So I think we may need to handle the unexpected response in 2 ways. If the correct response returns a result object, we can evaluate if a field exists, for example: https://github.com/linode/linode_api4-python/blob/dev/linode_api4/linode_client.py#L295
If the correct response returns nothing, we probably can follow the way that the base `delete` method does: https://github.com/linode/linode_api4-python/blob/dev/linode_api4/objects/base.py#L223 Instead of only returning the boolean indicator, we can still raise an error if it exists.

Reply:
Got it. Thanks for clearing that up!